From: Somnath Kotur <somnath.kotur@broadcom.com>
Date: Wed, 7 Sep 2016 19:57:50 +0530
Subject: be2net: Issue COMMON_RESET_FUNCTION cmd during driver unload
Patch-mainline: v4.9-rc1
Git-commit: f72099e057c0b3ea3cfd16301cff9202c4db8ef4
References: bsc#1021121 FATE#321675

As per SLI guideline, drivers need to issue COMMON_RESET_FUNCTION SLI
cmd during driver unload to clean up any non-persistent state
information.
Issue this cmd only if VFs are not assigned to VMs as it is possible
for PF driver to unload while it\'s VF remains functional and assigned
to a VM.

Signed-off-by: Somnath Kotur <somnath.kotur@broadcom.com>
Signed-off-by: Sriharsha Basavapatna <sriharsha.basavapatna@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/emulex/benet/be_main.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -5681,6 +5681,9 @@ static void be_remove(struct pci_dev *pd
 
 	be_clear(adapter);
 
+	if (!pci_vfs_assigned(adapter->pdev))
+		be_cmd_reset_function(adapter);
+
 	/* tell fw we're done with firing cmds */
 	be_cmd_fw_clean(adapter);
 
